### PR TITLE
perceptron cookbook

### PR DESCRIPTION
**Pull request by Saurabh7**

### CODE REVIEW

File: doc/cookbook/source/examples/classifier/perceptron.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
+=============================
```
Comment (karlnapf) [inline doc/cookbook/source/examples/classifier/perceptron.rst]: Please align

**Comment by karlnapf** [comment]
Integration test data?

File: doc/cookbook/source/examples/classifier/perceptron.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+Perceptron
+=============================
+
+The perceptron is an algorithm for binary classification. It maps its input :math:`x` to output value:math:`y` using the conditions:
+
+.. math::
+
+	y = \begin{cases}
+	  0, & \text{if } {\bf w}.x + b > 0, \\
```
Comment (karlnapf) [inline doc/cookbook/source/examples/classifier/perceptron.rst]: please use \bf{w}^\top x

File: doc/cookbook/source/examples/classifier/perceptron.rst [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	  1, & \text{otherwise}.
+	\end{cases}
+
+The algorithm allows for online learning by processing the elements in training set one at a time.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/classifier/perceptron.rst]: Say: It does not converge if the data is not linearly separable.

File: examples/meta/src/classifier/perceptron.sg [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+BinaryLabels labels_predict = perceptron.apply_binary(features_test)
+#![train_and_apply]
+
+#![extract_weights_bias]
+RealVector w = perceptron.get_w()
+real b = perceptron.get_bias()
+#![extract_weights_bias]
+
+#![evaluate_accuracy]
+AccuracyMeasure eval()
```
Comment (karlnapf) [inline examples/meta/src/classifier/perceptron.sg]: please use a different name, this is a python keyword

File: examples/meta/src/classifier/perceptron.sg [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+#![create_features]
+RealFeatures features_train(f_feats_train)
+RealFeatures features_test(f_feats_test)
+BinaryLabels labels_train(f_labels_train)
+BinaryLabels labels_test(f_labels_test)
+#![create_features]
+
+#![set_parameters]
+real learn_rate = 1
+real max_iter = 1000
```
Comment (karlnapf) [inline examples/meta/src/classifier/perceptron.sg]: maybe us 100 rather than 1000

File: examples/meta/src/classifier/perceptron.sg [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+#![create_features]
+RealFeatures features_train(f_feats_train)
+RealFeatures features_test(f_feats_test)
+BinaryLabels labels_train(f_labels_train)
+BinaryLabels labels_test(f_labels_test)
+#![create_features]
+
+#![set_parameters]
+real learn_rate = 1
```
Comment (karlnapf) [inline examples/meta/src/classifier/perceptron.sg]: Shall we maybe put the math for the updates so that we have a math analogue to the learning rate

**Comment by karlnapf** [comment]
You produced a type error in java:
https://travis-ci.org/shogun-toolbox/shogun/jobs/137217805#L2624

**Comment by karlnapf** [comment]
And csharp: https://travis-ci.org/shogun-toolbox/shogun/jobs/137217806#L1602

File: examples/meta/src/classifier/perceptron.sg (hunk 6 above)
Comment (sorig) [inline examples/meta/src/classifier/perceptron.sg]: Please update to:
```
real learn_rate = 1.0
int max_iter = 1000
```

Comment (Saurabh7) [inline examples/meta/src/classifier/perceptron.sg]: ok updated

File: examples/meta/src/binary_classifier/perceptron.sg [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+#![create_features]
+RealFeatures features_train(f_feats_train)
+RealFeatures features_test(f_feats_test)
+BinaryLabels labels_train(f_labels_train)
+BinaryLabels labels_test(f_labels_test)
+#![create_features]
+
+#![set_parameters]
+real learn_rate = 1.0
+real max_iter = 1000
```
Comment (sorig) [inline examples/meta/src/binary_classifier/perceptron.sg]: `int` on max_iter please :)

Comment (Saurabh7) [inline examples/meta/src/binary_classifier/perceptron.sg]: ah crapp :)

**Comment by Saurabh7** [comment]
Cool works now, will add data.

**Comment by karlnapf** [comment]
Merging as good

**Comment by karlnapf** [comment]
Needs data

**Pull request merged**